### PR TITLE
Fix Neon build warnings

### PR DESCRIPTION
**Pull request by cyb70289**

### CODE REVIEW

File: include/xsimd/arch/xsimd_neon.hpp [hunk 1]
```diff
@@ -1915,7 +1915,7 @@ namespace xsimd
         inline batch<T, A> extract_pair(batch<T, A> const& lhs, batch<T, A> const& rhs, std::size_t n, requires_arch<neon>) noexcept
         {
             constexpr std::size_t size = batch<T, A>::size;
-            assert(0 <= n && n < size && "index in bounds");
```
Comment (cyb70289) [inline include/xsimd/arch/xsimd_neon.hpp:1918]: warning as `n` is unsigned, always >= 0

File: include/xsimd/arch/xsimd_neon.hpp [hunk 2]
```diff
@@ -2053,7 +2053,7 @@ namespace xsimd
         template <class A, class T>
         inline batch<T, A> bitwise_lshift(batch<T, A> const& lhs, int n, requires_arch<neon>) noexcept
         {
-            constexpr std::size_t size = sizeof(typename batch<T, A>::value_type) * 8;
+            constexpr int size = sizeof(typename batch<T, A>::value_type) * 8;
             assert(0 <= n && n < size && "index in bounds");
             return detail::bitwise_lshift_impl(lhs, n, ::xsimd::detail::make_int_sequence<size>());
```
Comment (cyb70289) [inline include/xsimd/arch/xsimd_neon.hpp:2058]: Warning comparison of signed `n` and unsigned `size`.
Warning make_int_sequence<`size`> is unsigned, but the bitwise_lshift_impl expects `int`.

File: include/xsimd/arch/xsimd_scalar.hpp [hunk 3]
```diff
@@ -385,12 +385,12 @@ namespace xsimd
 
     inline int32_t nearbyint_as_int(float var) noexcept
     {
-        return std::nearbyint(var);
```
Comment (cyb70289) [inline include/xsimd/arch/xsimd_scalar.hpp:388]: waning float to int cast may change value

File: include/xsimd/math/xsimd_rem_pio2.hpp [hunk 4]
```diff
@@ -645,7 +645,7 @@ namespace xsimd
                     j = ix >> 20;
                     y[0] = r - w;
                     GET_HIGH_WORD(high, y[0]);
-                    i = j - ((high >> 20) & 0x7ff);
```
Comment (cyb70289) [inline include/xsimd/math/xsimd_rem_pio2.hpp:648]: `high` is unsigned, `i, j` are signed.

File: include/xsimd/math/xsimd_rem_pio2.hpp [hunk 5]
```diff
@@ -687,7 +687,7 @@ namespace xsimd
             GET_LOW_WORD(low, x);
             SET_LOW_WORD(z, low);
             e0 = (ix >> 20) - 1046; /* e0 = ilogb(z)-23; */
-            SET_HIGH_WORD(z, ix - ((std::int32_t)(e0 << 20)));
```
Comment (cyb70289) [inline include/xsimd/math/xsimd_rem_pio2.hpp:690]: `SET_HIGH_WORD` 2nd argument should be unsigned. `ix, e0` are signed.

**Comment by JohanMabille** [comment]
Thanks!

**Pull request merged**